weasyl/middleware.py

```diff
@@ -132,30 +163,52 @@ def __call__(self, environ, start_response):
 
 class SentryEnvironmentMiddleware(object):
     def __init__(self, app, dsn, reactor=None):
-        import raven
         self.app = app
-        self.client = raven.Client(dsn)
+        self.client = raven.Client(
+            dsn=dsn,
+            release=raven.fetch_git_sha(os.path.abspath(os.path.join(__file__, '../..'))),
+            processors=[
+                'raven.processors.SanitizePasswordsProcessor',
+                'weasyl.middleware.RemoveSessionCookieProcessor',
+            ],
+        )
         if reactor is None:
             from twisted.internet import reactor
         self.reactor = reactor
 
-    def ravenCaptureArguments(self, extra):
-        data = {}
-        if 'level' in extra:
-            data['level'] = extra.pop('level')
-        extra['env'] = dict(web.ctx.env)
-        extra['session'] = web.ctx.get('weasyl_session')
-        extra['deployed-revision'] = d.CURRENT_SHA
-        return dict(data=data, extra=extra)
+    def ravenCaptureArguments(self, level=None, **extra):
+        data = {
+            'level': level,
+            'user': {
+                'id': d.get_userid(),
+                'ip_address': d.get_address(),
+            },
+            'request': {
+                'url': web.ctx.env['PATH_INFO'],
+                'method': web.ctx.env['REQUEST_METHOD'],
+                'data': web.webapi.rawinput(method='POST'),
```

Comment:
Yes, it does – even for multipart/form-data. (Uploads make the requests non-replayable, but that’s really the best solution.)